xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/store/hibernate/query/VersioningStoreQueryFactory.java

```diff
@@ -93,11 +93,16 @@ private void applyCriteria(final long id, RevisionCriteria criteria)
             }
 
             Date minDate = criteria.getMinDate();
+            Date maxDate = criteria.getMaxDate();
             // Hibernate requires positive timestamps.
             if (minDate.getTime() < 0) {
                 minDate = new Date(0);
             }
-            predicates.add(this.builder.between(this.root.get(FIELD_DATE), minDate, criteria.getMaxDate()));
+            // Most databases store timestamps as seconds, using integers.
```

Comment:
This comment isn't clear. It should be improved as it's not obvious what the code is doing and why it works for all the databases we support